src/compile/scale/parse.ts

```diff
@@ -50,15 +52,21 @@ function parseUnitScaleCore(model: UnitModel): ScaleComponentIndex {
 
     const channelDef = encoding[channel];
 
+    // if mark is geoshape scale channel encodes the geojson
```

Comment:
This comment reads quite confusing.  

I think you mean "If mark is geoshape, we use projection instead of scale?"